ApeWorX/ape-solidity
# fix: issue where dependencies used local project's remappings instead of their own

### What I did

Somehow an unnoticed bug (until recently) where remappings were resolving to paths from the local projects listings rather than looking at config from the dependency's themselves.

### How I did it

Refactored all import remap stuff into its own module and relied on pydantic for some of the leg work.

### How to verify it

Easy. Make a project with OZ 4.5 and yearn vaults. Ensure yearn vaults imports resolve using OZ 4.7.1 since it is a dependenc of a dependency (different version of OZ). Lemme know if you need help, I can add more info.

### Checklist
